### ISSUE TITLE
Performance regression in `find_country_by_alpha2`

### ISSUE THREAD

**Issue by Gasparila** [hexorx/countries#415]
We recently upgraded from version 1.2.5 => 2.0.5 and noticed a ~10x slowdown in the library getting the country object by its codes using the `ISO3166::Country.find_country_by_alpha2` getter. We found that there's a better getter, new, but that just brings us back to where we were originally:

```
      alpha2   new
1.2.5|0.296104|0.003251
2.0.5|2.546312|0.297833
```

### Appendix: Script used and raw performance report

```
require 'benchmark'
country_codes = Port.order(id: :asc).limit(1000).pluck(:country_code)
Benchmark.bmbm do |bm|
  bm.report('find_by_alpha2') do
    country_codes.map { |cc| ISO3166::Country.find_country_by_alpha2 cc }
  end
  bm.report('new') do
    country_codes.map { |cc| ISO3166::Country.new cc }
  end
end
```
```
1.2.5 Benchmark
Rehearsal --------------------------------------------------
find_by_alpha2   0.350000   0.020000   0.370000 (  0.384616)
new              0.000000   0.000000   0.000000 (  0.003175)
----------------------------------------- total: 0.370000sec

                     user     system      total        real
find_by_alpha2   0.290000   0.000000   0.290000 (  0.296104)
new              0.000000   0.000000   0.000000 (  0.003251)

2.0.5 Benchmark
Rehearsal --------------------------------------------------
find_by_alpha2   2.520000   0.040000   2.560000 (  2.574010)
new              0.310000   0.010000   0.320000 (  0.325458)
----------------------------------------- total: 2.880000sec

                     user     system      total        real
find_by_alpha2   2.500000   0.030000   2.530000 (  2.546312)
new              0.300000   0.000000   0.300000 (  0.297833)
```

**Comment by rposborne** [comment]
Fantastic report @Gasparila,  did some spelunking and found a heavily used method calling `.uniq` on a large array. Check out #416 while not EXACTLY back up to speed, it is a 10x improvment.  

2.x does much more at runtime to ensure only the translations required are in memory.

**Comment by rposborne** [comment]
This has been merged and released under 2.0.6
